### PR TITLE
fix: apply x_stride1 in the rmsnorm normalize pass

### LINKED ISSUE #1467: [BUG] `_rms_norm_fwd_fused` drops `x_stride1` in its normalize pass, so a non-unit last-dim stride gives wrong output
## Summary

`lightllm/common/basemodel/triton_kernel/norm/rmsnorm.py` takes `x_stride1` and uses it in the variance pass, and the store uses `y_stride1`, but the normalize pass indexes the input as if it were contiguous:

```python
    for off in range(0, N, BLOCK_SIZE):
        cols = off + tl.arange(0, BLOCK_SIZE)
        x = tl.load(X + cols * x_stride1, mask=cols < N, other=0.0).to(tl.float32)   # line 32, honours the stride
        _var += x * x
    ...
    for off in range(0, N, BLOCK_SIZE):
        cols = off + tl.arange(0, BLOCK_SIZE)
        mask = cols < N
        ...
        x = tl.load(X + cols, mask=mask, other=0.0).to(tl.float32)                   # line 42, does not
        ...
        tl.store(Y + cols * y_stride1, y.to(Y.dtype.element_ty), mask=mask)          # line 48, honours it
```

So whenever `x.stride(-1) != 1` the row's RMS is computed from the correct elements while the values that get scaled by it are read from somewhere else — for a 2-D transposed input, straight down a column. The result is silently wrong; there is no error and nothing goes out of bounds.

`rmsnorm_forward` has no guard against this. It does `x_arg = x.view(-1, x.shape[-1])`, which for a 2-D input is shape-preserving and therefore succeeds whatever the strides are, and `torch.empty_like(x)` gives the output the same strides, so `y_stride1` matches and only the load is wrong.

The three in-tree call sites (`gemma4`, `deepseek3_2`, `NormWeight`) all pass tensors whose last-dim stride is 1 today — `deepseek3_2` slices `cache_kv[:, :, :kv_lora_rank]`, which keeps stride 1 on the last dim — so this does not currently break a shipped model. But the parameter exists to support strided inputs, two of the three places that use it get it right, and any caller that passes one gets wrong numbers rather than a complaint.

Introduced with the kernel; the file was last touched in #1304.

## Reproduction

```python
import torch
from lightllm.common.basemodel.triton_kernel.norm.rmsnorm import rmsnorm_forward, torch_rms_norm

M, N = 64, 256
torch.manual_seed(0)
weight = torch.rand(N, dtype=torch.float32, device="cuda")

x = torch.randn(N, M, device="cuda").t()          # shape (M, N), stride (1, M)
out = rmsnorm_forward(x, weight, 1e-6)
ref = torch_rms_norm(x.float(), weight.float(), 1e-6)
print("strided   max|err|:", (out.float() - ref).abs().max().item())

xc = x.contiguous()                                # identical values, stride (N, 1)
out = rmsnorm_forward(xc, weight, 1e-6)
ref = torch_rms_norm(xc.float(), weight.float(), 1e-6)
print("contiguous max|err|:", (out.float() - ref).abs().max().item())
```

```
strided   max|err|: 5.832374572753906
contiguous max|err|: 0.0
```

15801 of 16384 output elements are wrong in the strided case (16230 with `weight=None`).

## Expected Behavior

The normalize pass reads with `x_stride1`, like the variance pass above it and the store below it.

## Environment

- `ModelTC/lightllm` at `fe9bdabfc331b990124f1ec27daf6bb7945cf7ee`
- NVIDIA B200 (sm_100), driver 595.71.05
- torch 2.13.0+cu130, triton 3.7.1, Python 3.12

### PR DESCRIPTION
**Pull request by truong-v**
Fixes #1467

`_rms_norm_fwd_fused` honours `x_stride1` when it accumulates the variance, and honours `y_stride1` when it stores, but the normalize pass reads `X + cols`. For any input whose last-dim stride is not 1 the row's RMS is therefore computed from the right elements while the values it scales come from elsewhere, and the output is silently wrong.

```diff
-        x = tl.load(X + cols, mask=mask, other=0.0).to(tl.float32)
+        x = tl.load(X + cols * x_stride1, mask=mask, other=0.0).to(tl.float32)
```

`rmsnorm_forward` reaches the kernel with such a tensor without complaint: `x.view(-1, x.shape[-1])` is shape-preserving for a 2-D input and succeeds whatever the strides are, and `torch.empty_like(x)` gives the output matching strides.

The `gemma4`, `deepseek3_2` and `NormWeight` call sites all pass last-dim-contiguous tensors today, so no shipped model is affected — `deepseek3_2`'s `cache_kv[:, :, :kv_lora_rank]` slice keeps stride 1 on the last dim. This makes the kernel consistent with the strides it already accepts.

## Testing

New `unit_tests/common/basemodel/triton_kernel/test_rmsnorm.py`, 12 cases (3 shapes × with/without weight × contiguous/strided). All 12 pass with this change. Reverting the kernel and keeping the tests fails the 6 strided cases and passes the 6 contiguous ones.

Against `torch_rms_norm` from the same module, at `M=64, N=256` float32:

| input | before | after |
|---|---|---|
| contiguous, weight | 0.0 | 0.0 |
| contiguous, no weight | 0.0 | 0.0 |
| transposed (last-dim stride 64), weight | 5.83 (15801/16384 cells wrong) | 0.0 |
| transposed (last-dim stride 64), no weight | 6.58 (16230/16384 cells wrong) | 0.0 |
| the same values, made contiguous | 0.0 | 0.0 |

## Performance

`triton.testing.do_bench` on an NVIDIA B200, three interleaved before/after rounds:

| shape | before | after |
|---|---|---|
| M=4096 N=4096 fp16 | 20.48 / 20.52 / 20.48 us | 20.47 / 20.45 / 20.48 us |
| M=8192 N=8192 fp16 | 51.21 / 51.21 / 51.21 us | 51.21 / 51.21 / 51.20 us |
| M=16384 N=2048 bf16 | 34.92 / 34.93 / 34.93 us | 34.95 / 34.93 / 34.91 us |
| M=1024 N=7168 bf16 | 12.42 / 12.44 / 12.43 us | 12.45 / 12.44 / 12.43 us |

No measurable cost — the kernel is memory-bound and the variance pass already does the same multiply.

`black --line-length=120` and `flake8` with the repo's pre-commit arguments are clean on both files.

## Environment

- `ModelTC/lightllm` at `fe9bdabfc331b990124f1ec27daf6bb7945cf7ee`
- NVIDIA B200 (sm_100), driver 595.71.05
- torch 2.13.0+cu130, triton 3.7.1, Python 3.12

### CODE REVIEW

**Review by holtwood — COMMENTED** [review]
Could you remove `lightllm/common/basemodel/triton_kernel/fused_moe/deepep_scatter_gather.py` from this PR? It looks unrelated to #1467 and the RMSNorm stride fix. Keeping this PR to the kernel fix and its regression test would make it easier to review.
